src/hotspot/cpu/aarch64/c2_MacroAssembler_aarch64.cpp

```diff
@@ -944,3 +944,56 @@ void C2_MacroAssembler::neon_compare(FloatRegister dst, BasicType bt, FloatRegis
     }
   }
 }
+
+void C2_MacroAssembler::sve_compare(PRegister pd, BasicType bt, PRegister pg,
+                                    FloatRegister zn, FloatRegister zm, int cond) {
+  assert(pg->is_governing(), "This register has to be a governing predicate register");
+  FloatRegister z1 = zn, z2 = zm;
+  // Convert the original BoolTest condition to Assembler::condition.
+  Condition condition;
+  switch (cond) {
+    case BoolTest::eq: condition = Assembler::EQ; break;
+    case BoolTest::ne: condition = Assembler::NE; break;
+    case BoolTest::le: z1 = zm; z2 = zn;  // Swap the two inputs
```

Comment:
This might warn on some targets, so we'd need to fix it anyway, so please do it now. Better to duplicate the action and break.

Reply:
Addressed in the new commit. Thanks!